shell/platform/windows/keyboard_key_handler.cc

```diff
@@ -96,10 +178,39 @@ bool KeyboardKeyHandler::KeyboardHook(FlutterWindowsView* view,
     return false;
   }
 
+  if (IsKeyDownAltRight(action, key, extended)) {
+    if (last_key_is_ctrl_left_down) {
+      should_synthesize_ctrl_left_up = true;
+    }
+  }
+  if (IsKeyDownCtrlLeft(action, key)) {
+    last_key_is_ctrl_left_down = true;
+    ctrl_left_scancode = scancode;
+    should_synthesize_ctrl_left_up = false;
+  } else {
+    last_key_is_ctrl_left_down = false;
+  }
+  if (IsKeyUpAltRight(action, key, extended)) {
+    if (should_synthesize_ctrl_left_up) {
+      should_synthesize_ctrl_left_up = false;
+      PendingEvent ctrl_left_up{
+          .key = VK_LCONTROL,
+          .scancode = ctrl_left_scancode,
+          .action = WM_KEYUP,
+          .was_down = true,
+      };
+      DispatchEvent(ctrl_left_up);
+    }
+  }
+
   uint64_t sequence_id = ++last_sequence_id_;
   incoming->sequence_id = sequence_id;
   incoming->unreplied = delegates_.size();
-  incoming->any_handled = false;
+  // There are a few situations where events must not be redispatched.
+  // Initializing `any_handled` with true in such cases suffices since it can
+  // only be set true and is used to disable redispatching.
```

Comment:
```suggestion
  // Initializing `any_handled` with true in such cases suffices, since it can
  // only be set to true and is used to disable redispatching.
```